pkg/operator/operator.go

```diff
@@ -133,6 +137,34 @@ func New(ctx context.Context, config operatorconfig.Config, kubeConfig *rest.Con
 		return nil, fmt.Errorf("failed to create status controller: %v", err)
 	}
 
+	// Set up the DNSNameResolver controller. This controller is unmanaged by
+	// the manager; the enable_dnsnameresolver controller starts it and the
+	// caches.
+	dnsNameResolverController, dnsNameResolverControllerCaches, err := dnsnameresolver.NewUnmanaged(operatorManager, dnsnameresolver.Config{
```

Comment:
Please mention the two caches represented in `dnsNameResolverCaches`: `dnsNameResolverCache` and `corednsEndpointsSliceCache`.